scripts/check_formatting

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+
+# Runs clang-format and rustfmt on the entire codebase to check if it's 
+# well formatted. Exit codes:
+#  - 1 there are files to be formatted.
+#  - 0 everything looks fine.
+
+# We ask clang-format to generate an XML output with the deisred formattings.
+# We check if any "replacement" tag exists in the output.
+CLANG_FORMAT_EXIT_CODE=0
+find oak examples -type f -name '*.h' -o -name '*.cc' -o -name '*.proto' \
+  | xargs -I{} clang-format -style=file -output-replacements-xml {} \
+  | grep "<replacement " >/dev/null \
+  && CLANG_FORMAT_EXIT_CODE=$? || CLANG_FORMAT_EXIT_CODE=$?
+
+# Fortunately, rustfmt has the --check option that will make it exit with 1
+# if formatting has to be applied.
+RUSTFMT_EXIT_CODE=0
```

Comment:
May be easier to just stop at the first command that errors out, rather than try to collect exit codes? Because we already set `errexit` this should be as simple as just running the commands by themselves (well, including the grep part for `clang-format` I guess)

Reply:
I thought it is more explicit like this, as the condition with clang-tidy has a nasty twist (abusing grep behavior), and this way in the end the if-else makes clear what happened when. Also rustfmt gives you a nice diff output so you know what's not formatted (unfortunately clang-tidy doesn't), so I thought it's worth running regardless clang-tidy has failed or not.

Reply:
In practice no one would look at the diff anyways, they would just run the format script :) and keeping around the extra logic makes it more complicated to add more linter in the future (e.g. I would like to add clang-tidy and something for BUILD files, but I don't think it's worth trying to follow this pattern for all of them, vs just failing when the first one of them fails).

Reply:
OK, I simplified it, I also suppressed the output of rustfmt --check.

Reply:
Ah, I just realized that the script this way will not print anything if the formatting is insufficient. For CI use it's probably fine (?), but if you ever wanna add this as a pre-commit hook or use manually it will be confusing. Given this I'd rather restore the previous state.